### PR TITLE
Random seek end command ,  seek support for parseFS disabled case and read timestamp from video header

### PR DESCRIPTION
**Pull request by mohammedzakikochargi**
**IMPORTANT: All PRs must be linked to an issue (except for extremely trivial and straightforward changes).**

Fixes #[Issue]

**Description : Random seek optional end command . It'll help to seek till a specific timeStamp in a video file , it is a optional command , if not given then it seeks till the end of video file.

Precise description of the changes in this pull request

**Alternative(s) considered**

Have you considered any alternatives? And if so, why have you chosen the approach in this PR?

**Type**

Type Choose one: ( Feature )

**Screenshots (if applicable)**

**Checklist**

- [x] I have read the [Contribution Guidelines](https://github.com/Apra-Labs/ApraPipes/wiki/Contribution-Guidelines)
- [x] I have written Unit Tests
- [x] I have discussed my proposed solution with code owners in the linked issue(s) and we have agreed upon the general approach

### CODE REVIEW

**Review by kumaakh — CHANGES_REQUESTED** [review]
Do we need to commit the 77 MB mp4 file ? can it be avoided ?

File: base/src/Mp4ReaderSource.cpp [hunk 1]
```diff
@@ -372,6 +444,12 @@ void Mp4readerDetailJpeg::setMetadata()
 	return;
 }
 
+int Mp4readerDetailJpeg::mp4Seek(mp4_demux* demux, uint64_t time_offset_usec, mp4_seek_method syncType) 
+{
+	auto ret = mp4_demux_seek_jpeg(demux, time_offset_usec, syncType, &seekedToFrame);
```
[the remark this answers is not in the document]
Comment (mohammedzakikochargi) [reply base/src/Mp4ReaderSource.cpp:449]: Yes , we check for check for success or failure of seek at the place of function call.

File: base/src/Mp4ReaderSource.cpp [hunk 2]
```diff
@@ -427,6 +516,12 @@ void Mp4readerDetailH264::setMetadata()
 	return;
 }
 
+int Mp4readerDetailH264::mp4Seek(mp4_demux* demux, uint64_t time_offset_usec, mp4_seek_method syncType)
+{
+	auto ret = mp4_demux_seek(demux, time_offset_usec, syncType, &seekedToFrame);
```
[the remark this answers is not in the document]
Comment (mohammedzakikochargi) [reply base/src/Mp4ReaderSource.cpp:521]: ret is required to check is the seek has happened successfully.

File: base/src/Mp4ReaderSource.cpp [hunk 3]
```diff
@@ -121,13 +123,23 @@ class Mp4readerDetailAbs
...
 			LOG_INFO << "Reading User Metadata Key-Values\n";
 			for (auto i = 0; i < count; i++) {
-				if ((keys[i]) && (values[i]) && !strcmp(keys[i], "\251too"))
+				if ((keys[i]) && (values[i]))
 				{
-					LOG_INFO << "key <" << keys[i] << ",<" << values[i] << ">";
-					mState.mSerFormatVersion.assign(values[i]);
+					if (!strcmp(keys[i], "\251too"))
+					{
+						LOG_INFO << "key <" << keys[i] << ",<" << values[i] << ">";
```
[the remark this answers is not in the document]
Comment (mohammedzakikochargi) [reply base/src/Mp4ReaderSource.cpp:135]: Not required , made it as log debug

File: base/src/Mp4ReaderSource.cpp [hunk 4]
```diff
@@ -121,13 +123,23 @@ class Mp4readerDetailAbs
...
+						mState.mSerFormatVersion.assign(values[i]);
+					}
+					if (!strcmp(keys[i], "\251sts"))
+					{
+						LOG_INFO << "key <" << keys[i] << ",<" << values[i] << ">";
```
[the remark this answers is not in the document]
Comment (mohammedzakikochargi) [reply base/src/Mp4ReaderSource.cpp:140]: No really , made it as log info

File: base/src/Mp4ReaderSource.cpp [hunk 5]
```diff
@@ -159,16 +171,69 @@ class Mp4readerDetailAbs
...
+			{
+				LOG_INFO << "seek time outside range. Seeking to start of video.";
+				skipMsecsInFile = 0;
+			}
+			else
+			{
+				skipMsecsInFile = skipTS - mState.startTimeStamp;
+			}
+
+			LOG_INFO << "Attempting seek <" << mState.mVideoPath << "> @skipMsecsInFile <" << skipMsecsInFile << ">";
```
[the remark this answers is not in the document]
Comment (mohammedzakikochargi) [reply base/src/Mp4ReaderSource.cpp:220]: Made it log debug

**Pull request merged**